pkg/asset/cluster/cluster.go

```diff
@@ -97,31 +91,14 @@ func (c *Cluster) Generate(parents asset.Parents) (err error) {
 		return errors.New("cluster cannot be created with bootstrapInPlace set")
 	}
 
-	// Check if we're using Cluster API.
-	if capiutils.IsEnabled(installConfig) {
-		// TODO(vincepri): The context should be passed down from the caller,
-		// although today the Asset interface doesn't allow it, refactor once it does.
-		ctx, cancel := context.WithCancel(signals.SetupSignalHandler())
-		go func() {
-			<-ctx.Done()
-			cancel()
-			clusterapi.System().Teardown()
-		}()
-
-		return c.provisionWithClusterAPI(ctx, parents, installConfig, clusterID)
-	}
-
-	// Otherwise, use the normal path.
-	return c.provision(installConfig, clusterID, terraformVariables)
-}
-
-func (c *Cluster) provision(installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID, terraformVariables *TerraformVariables) error {
 	platform := installConfig.Config.Platform.Name()
 
 	if azure := installConfig.Config.Platform.Azure; azure != nil && azure.CloudName == typesazure.StackCloud {
 		platform = typesazure.StackTerraformName
 	}
 
+	// TODO(padillon): determine whether CAPI handles tagging shared subnets, in which case we should be able
+	// to encapsulate these into the terraform package.
```

Comment:
Probably not today, I can double check, but we should open an issue upstream if not